### ISSUE TITLE
Implementing Multitenancy

### ISSUE THREAD

**Issue by kirandeshmukh** [joshsoftware/the_hunger_terminal#140]
Now, we are using this software for Josh Software only. Our aim is to `make this software multi tenant.`
In multi tenancy, the main problem is **cross platform Data Leaks**.  For implementing this feature, there are following  available gems:

1.Multitenant
2.Apartment
3.activerecord-multi-tenant

Apartment gem is useful for multi tenancy. It still allows us to have some data in a common tenant. I think, this `common tenant `is not required in The Hunger Terminal. Activerecord-multi-tenant gem is useful for `distributed multi-tenant databases` - which is not of our concern . So, I think we should prefer Multitenant gem in our application.

**Comment by gautamrege** [comment]
Yes. Also, verify with Anil and Shailesh as they have used a lot of multi tenancy

Ignore branch offices for now. We shall taraise it when the need arises
